### PR TITLE
load project groups to DW

### PR DESCRIPTION
**Pull request by gondzo**
Before merging, a new table should be created in DW:tcs_dw

`create table project_groups (project_id DECIMAL(12,0), group_id DECIMAL(12,0), PRIMARY KEY (project_id, group_id) CONSTRAINT project_group_pkey);`

### CODE REVIEW

File: src/main/com/topcoder/utilities/dwload/TCLoadTCS.java [hunk 1]
```diff
@@ -2750,6 +2752,93 @@ public void doLoadProjectTechnologies() throws Exception {
...
+            rs.next();
+
+            // no records, it's the first run of loading groups
+            boolean firstRun = rs.getInt(1) == 0;
+
+            if(firstRun) log.info("Loading project group table for the first time. A complete load will be performed");
+
+            final String SELECT = "select p.project_id, group_id from project p INNER JOIN common_oltp:contest_eligibility ce ON ce.contest_id = p.project_id INNER JOIN common_oltp:group_contest_eligibility gce ON gce.contest_eligibility_id=ce.contest_eligibility_id \n" +
+                    (firstRun ? "" : " AND (p.create_date > ? OR p.modify_date > ?)") +
+                    "group by p.project_id, group_id";
```
Comment (RishiRajSahu) [inline src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: @gondzo what is the purpose of grouping here ? Also, though it seems to be correct still verify manually by running the query that we get record for each group project  is part of as project could be part of more than one group and no group at all.

File: src/main/com/topcoder/utilities/dwload/TCLoadTCS.java [hunk 2]
```diff
@@ -2750,6 +2752,93 @@ public void doLoadProjectTechnologies() throws Exception {
...
+            rs = selectGroups.executeQuery();
+
+            int countRecords = 0;
+
+            while (rs.next()) {
+                long projectID = rs.getLong("project_id");
+                long groupId = rs.getLong("group_id");
+
+                if(!firstRun && !deletedProjects.contains(projectID)) {
+                    // the load is not run for the first time && it's not processed in this load, clear the old technologies for the project
```
Comment (RishiRajSahu) [inline src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: @gondzo please update the comment 'technologies'

File: src/main/com/topcoder/utilities/dwload/TCLoadTCS.java (hunk 1 above)
Comment (gondzo) [reply src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: the grouping is added as some projects have duplicate groups in the db - there is no unique key on project_id, group_id, so grouping just avoids that case

File: src/main/com/topcoder/utilities/dwload/TCLoadTCS.java (hunk 2 above)
Comment (gondzo) [reply src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: done

File: src/main/com/topcoder/utilities/dwload/TCLoadTCS.java (hunk 1 above)
Comment (RishiRajSahu) [reply src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: @gondzo got it, but could you verify the second point, when a project is associated to multiple groups, will be having multiple records for that in our target DW table ?

File: src/main/com/topcoder/utilities/dwload/TCLoadTCS.java [hunk 3]
```diff
@@ -2750,6 +2752,93 @@ public void doLoadProjectTechnologies() throws Exception {
...
+            selectGroups = prepareStatement(SELECT, SOURCE_DB);
+
+            if(!firstRun) {
+                // no the first time, set last loading time
+                selectGroups.setTimestamp(1, fLastLogTime);
+                selectGroups.setTimestamp(2, fLastLogTime);
+            }
+
+
+            final String DELETE = "delete from project_groups where project_id = ?";
```
Comment (ajefts) [inline src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: @RishiRajSahu @gondzo Just making sure we're intending to delete ALL groups for the given project here as opposed to a specific project + group record?

Comment (gondzo) [reply src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: yes, the idea is to delete all and add them back when a project is updated (same thing is done for project platforms, technologies, etc)

File: src/main/com/topcoder/utilities/dwload/TCLoadTCS.java (hunk 1 above)
Comment (gondzo) [reply src/main/com/topcoder/utilities/dwload/TCLoadTCS.java]: yes, multiple records will be created in the target DW table

**Pull request merged**